drinkingfountain/src/main/java/sample/knot/cesar/org/br/drinkingfountain/adapter/DrinkFountainGridAdapter.java

```diff
@@ -0,0 +1,93 @@
+package sample.knot.cesar.org.br.drinkingfountain.adapter;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+import sample.knot.cesar.org.br.drinkingfountain.R;
+import sample.knot.cesar.org.br.drinkingfountain.database.FacadeDatabase;
+import sample.knot.cesar.org.br.drinkingfountain.model.DrinkFountainDevice;
+import sample.knot.cesar.org.br.drinkingfountain.model.WaterLevelData;
+import sample.knot.cesar.org.br.drinkingfountain.view.WaterBottleView;
+
+/**
+ * Created by usuario on 20/01/17.
+ */
+
+public class DrinkFountainGridAdapter extends RecyclerView.Adapter<DrinkFountainGridAdapter.WaterViewHolder> {
+
+    private Context mContext;
+    private List<DrinkFountainDevice> mListOfWaterBottle;
+    private FacadeDatabase mDataBase;
+
+    public DrinkFountainGridAdapter(Context context, List<DrinkFountainDevice> list) {
+        mContext = context;
+        mListOfWaterBottle = list;
+        mDataBase = FacadeDatabase.getInstance();
+
+    }
+
+    @Override
+    public WaterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+
+        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.water_card_view, parent, false);
+
+        return new WaterViewHolder(itemView);
```

Comment:
onCreateViewHolder only creates a new view holder when there are no existing view holders which the RecyclerView can reuse. So, for instance, if your RecyclerView can display 5 items at a time, it will create 5-6 ViewHolders, and then automatically reuse them, each time calling onBindViewHolder.

Its similar to what your code in the ListView does (checking if convertView is null, and if not, grabbing the existing ViewHolder from the tag), except, with RecyclerView, this is all done automatically.

I suppose this is one of the pros with using a RecyclerView - you don't need to worry so much about reusing ViewHolders as you do with ListView. The con is, RecyclerView is very customisable, but has very little built in functionality - unlike ListView which is not very customisable, but has a lot of built in functionality.